packages/documentsearch-extension/src/index.ts

```diff
@@ -0,0 +1,367 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+import '../style/index.css';
+
+import { SearchProviderRegistry } from './searchproviderregistry';
+
+import {
+  JupyterLab,
+  JupyterLabPlugin,
+  ApplicationShell
+} from '@jupyterlab/application';
+import { ICommandPalette } from '@jupyterlab/apputils';
+
+import { ISignal, Signal } from '@phosphor/signaling';
+import { createSearchOverlay } from './searchoverlay';
+import { Widget } from '@phosphor/widgets';
+import { DocumentWidget } from '@jupyterlab/docregistry';
+
+export interface ISearchMatch {
+  /**
+   * Text of the exact match itself
+   */
+  readonly text: string;
+
+  /**
+   * Fragment containing match
+   */
+  readonly fragment: string;
+
+  /**
+   * Line number of match
+   */
+  line: number;
+
+  /**
+   * Column location of match
+   */
+  column: number;
+
+  /**
+   * Index among the other matches
+   */
+  index: number;
+}
+
+export interface ISearchProvider {
+  /**
+   * Initialize the search using the provided options.  Should update the UI
+   * to highlight all matches and "select" whatever the first match should be.
+   *
+   * @param options All of the search parameters configured in the search panel
```

Comment:
There is no 'options' argument for this function. Can you instead explain `query` and `searchTarget`?